cmd/zdb/zdb.c

```diff
@@ -796,38 +831,125 @@ dump_metaslab(metaslab_t *msp)
 static void
 print_vdev_metaslab_header(vdev_t *vd)
 {
-	(void) printf("\tvdev %10llu\n\t%-10s%5llu   %-19s   %-15s   %-10s\n",
-	    (u_longlong_t)vd->vdev_id,
+	char class[32] = "(";
+
+	if (vd->vdev_islog || vd->vdev_mg->mg_class[SPA_CLASS_LOG] != NULL)
+		(void) strcat(class, VDEV_CLASS_LOG",");
+	if (vd->vdev_mg->mg_class[SPA_CLASS_MOS] != NULL)
+		(void) strcat(class, VDEV_CLASS_MOS",");
+	if (vd->vdev_mg->mg_class[SPA_CLASS_DMU] != NULL)
+		(void) strcat(class, VDEV_CLASS_DMU",");
+	if (vd->vdev_mg->mg_class[SPA_CLASS_DDT] != NULL)
+		(void) strcat(class, VDEV_CLASS_DDT",");
+
+	if (strlen(class) == 1)
```

Comment:
Maybe this, for consistency with later code (and avoiding calling strlen() twice)? `if (class[1] != '\0')`